Cagiva Chaedar Bey Lirna

# Permasalahan

A company wants to run a new campaign to increase the transactions’ amount of its product.
Before running the actual campaign to the whole customers, the company wants to know the 
campaign’s effectiveness in their 30 days transactions’ amount.
The average of transactions’ amount before the campaign is $100.
The company ran a testing on 20 customers and gathered their transactions’ amount in the next 30 
days after the campaign. The data is stored in this table below.
Run a t-test analysis on the transactions’ amount average (with alpha = 5%). 

# Proses mencari solusi

Campign dikatakan efektif jika jumlah transaksi selama masa percobaan 30 hari lebih besar dibanding dengan jumlah transaksi sebelum campign. Rata-rata jumlah transaksi sebelum campign adalah $100.

### 1. Menentukan Hipotesis

H0 : Mean jumlah transaksi selama masa percobaan 30 hari = `$100`

H1 : Mean jumlah transaksi selama masa percobaan 30 hari > `$100`

### 2. Menentukan alpha (Tingkat Signifikansi)

In [29]:
alpha = 0.05

### 3. Uji - t

In [30]:
import scipy.stats as stats
import pandas as pd
import numpy as np

In [31]:
jml_transaksi= np.array([100, 150, 50, 100, 130, 120, 100, 110, 75, 65, 150, 120, 50, 100, 100, 140, 90, 150, 50, 90])

In [32]:
tstatistics, pvalue = stats.ttest_1samp(a=jml_transaksi, popmean=100)

In [35]:
def kesimpulan():
    df_transaksi= pd.DataFrame(jml_transaksi, columns=["Jumlah Transaksi"])
    print("Data yang dianalisis\n-----------------------------")
    print(df_transaksi, "\n-----------------------------")
    print("Menghitung nilai t - statistics dan p - value","\na. t - statistics =", tstatistics, "\nb. p - value =", pvalue)
    critical_tvalues = stats.t.ppf(1-alpha, 19)
    print("Setelah mendapat nilai t-statistics dan p-value, selanjutnya hitung critical T-values untuk menyimpulkan hipotesis yang dibuat.\n-----------------------------")
    print("critical T - values  =", critical_tvalues, "\n-----------------------------", "\nKesimpulan")
    if tstatistics<critical_tvalues:
        print('t-statistics = %.2f'%tstatistics, '<', 'critical T-values = %.2f'%critical_tvalues, '\nKarena t-statistics < Critical T-values, maka hipotesis H0 diterima.')
    else:
        print('t-statistics = %.2f'%tstatistics, '>', 'critical T-values = %.2f'%critical_tvalues,'\nKarena t-statistics > Critical T-values, maka hipotesis H1 diterima.')

# Solusi dan Kesimpulan

In [36]:
kesimpulan()

Data yang dianalisis
-----------------------------
    Jumlah Transaksi
0                100
1                150
2                 50
3                100
4                130
5                120
6                100
7                110
8                 75
9                 65
10               150
11               120
12                50
13               100
14               100
15               140
16                90
17               150
18                50
19                90 
-----------------------------
Menghitung nilai t - statistics dan p - value 
a. t - statistics = 0.2731657711666385 
b. p - value = 0.787672894169515
Setelah mendapat nilai t-statistics dan p-value, selanjutnya hitung critical T-values untuk menyimpulkan hipotesis yang dibuat.
-----------------------------
critical T - values  = 1.729132811521367 
----------------------------- 
Kesimpulan
t-statistics = 0.27 < critical T-values = 1.73 
Karena t-statistics < Critical T-values, maka hipotesis H0 diterima

Karena didapat nilai t - statistics < critical T-values (0.27 < 1.73), sehingga dapat disimpulkan bahwa hipotesis H0 diterima atau rata-rata jumlah transaksi selama masa percobaan 30 hari = $100. Perlu digarisbawahi juga bahwa campaign yang dilakukan selama 30 hari tidak efektif dalam meningkatkan jumlah transaksi.